mirrord/layer-lib/src/process/windows/execution/pipes.rs

```diff
@@ -0,0 +1,289 @@
+        if let Some(handle) = self.forwarding_thread.take() {
+            unsafe {
+                CloseHandle(self.read_handle);
+                self.read_handle = NULL;
+            }
+
+            let _ = handle.join();
+        }
+    }
+
+    /// Detach forwarding thread to continue running independently
+    /// This allows the forwarding to continue even after this object is dropped
+    pub fn detach_forwarding(&mut self) {
+        if let Some(handle) = self.forwarding_thread.take() {
+            // Don't join the thread, let it continue running independently
+            // The thread will naturally exit when the child process closes its end of the pipe
+            std::mem::forget(handle);
+
+            // Don't close handles here - let the child process continue using them
+            // The handles will be cleaned up when the child process exits
+        }
+    }
+}
+
+impl Drop for PipeForwarder {
+    fn drop(&mut self) {
+        self.stop_forwarding();
+
+        unsafe {
+            if self.read_handle != NULL {
+                CloseHandle(self.read_handle);
+            }
+            if self.write_handle != NULL {
+                CloseHandle(self.write_handle);
+            }
+        }
+    }
+}
+
+/// Manages the three pipe forwarders for stdio redirection
+pub struct StdioRedirection {
+    pub stdout: Option<PipeForwarder>,
+    pub stderr: Option<PipeForwarder>,
+    // Future expansion
+    pub stdin: Option<PipeForwarder>,
+}
+
+impl StdioRedirection {
+    /// Create stdio redirection with stdout/stderr pipes
+    pub fn create_for_child() -> LayerResult<Self> {
+        Ok(Self {
+            stdout: Some(PipeForwarder::new_anonymous("stdout")?),
+            stderr: Some(PipeForwarder::new_anonymous("stderr")?),
+            stdin: None, // Future expansion
+        })
+    }
+
+    /// Apply operation to both stdout and stderr pipes
+    fn apply_to_pipes_void<F>(&mut self, mut operation: F)
+    where
+        F: FnMut(&mut PipeForwarder),
+    {
+        if let Some(ref mut stdout) = self.stdout {
+            operation(stdout);
+        }
+        if let Some(ref mut stderr) = self.stderr {
+            operation(stderr);
+        }
+    }
+
+    /// Setup startup info with pipe handles for child process
+    pub fn setup_startup_info(&self, startup_info: &mut STARTUPINFOW) {
+        startup_info.dwFlags |= STARTF_USESTDHANDLES;
+
+        if let Some(ref stdout) = self.stdout {
+            startup_info.hStdOutput = stdout.write_handle();
+        }
+        if let Some(ref stderr) = self.stderr {
+            startup_info.hStdError = stderr.write_handle();
+        }
+    }
+
+    /// Start forwarding stdout/stderr to parent handles
+    pub fn start_forwarding(
+        &mut self,
+        parent_stdout: HANDLE,
+        parent_stderr: HANDLE,
+    ) -> LayerResult<()> {
+        tracing::debug!(
+            "🎯 StdioRedirection::start_forwarding called with stdout: {:?}, stderr: {:?}",
```

Comment:
let's please remove the emojis, they're super inconsistent, and not every font/terminal emulator supports this charset